book/src/configuration.md

```diff
@@ -49,6 +49,7 @@ on unix operating systems.
 | `gutters` | Gutters to display: Available are `diagnostics` and `line-numbers` and `spacer`, note that `diagnostics` also includes other features like breakpoints, 1-width padding will be inserted if gutters is non-empty | `["diagnostics", "line-numbers"]` |
 | `auto-completion` | Enable automatic pop up of auto-completion. | `true` |
 | `auto-format` | Enable automatic formatting on save. | `true` |
+| `auto-save` | Enable automatic saving on focus moving away from Helix. Requires [focus event support](https://github.com/helix-editor/helix/wiki/Terminal-Support) from your terminal. | `false` |
```

Comment:
I think it would be better to have this as an array to prevent breaking config breakage if new auto save events are added, like you mentioned in the opening comment:

>  ...I think an option set like the gutters option could make sense. For this, we'd have an option autosave with a default value of ["focus-lost"]...

Reply:
This doesn't have to be a breaking change. The type could just become an enum where the value is allowed to be either a boolean or a list of strings. This is what the auto pairs config option does. True could just translate to the default list internally.

Reply:
Yeah, having it function like `auto-pairs` for expansion was my rough thinking. Happy to change to a list now if that's the preferred way.